src/app/util/attribute-storage.cpp

```diff
@@ -1122,6 +1122,10 @@ void emAfLoadAttributeDefaults(EndpointId endpoint, bool ignoreStorage)
     uint16_t epCount = emberAfEndpointCount();
     uint8_t attrData[ATTRIBUTE_LARGEST];
     auto * attrStorage = ignoreStorage ? nullptr : app::GetAttributePersistenceProvider();
+    if (!ignoreStorage && !attrStorage)
+    {
+        ChipLogProgress(DataManagement, "Attribute persistence needs a persistence provider");
```

Comment:
I disagree with these types of runtime-only logs that are meant to disappear once you integrate properly.

Really, if the framework uses a persistence provider, we need to early-on "break" (log + crash), e.g. in Server.cpp or some init function of this module if persistence is needed and we don't have it

Reply:
I agree that would be better, but I don't quite know where to do that....

Realistically, `emAfLoadAttributeDefaults` is in fact part of the init path for this module, so one option would be to just log and crash here.  Thoughts?

Reply:
Leaving this as-is until we come up with a better idea....